### PR TITLE
gitAndTools.qgit: 2.8 -> 2.9

### PR DESCRIPTION
**Pull request by markuskowa**
###### Motivation for this change
Regular update

###### Things done
Cleanup/modernize:

* switch`name` -> `pname`
* qtApps compatibility: use Qt's `mkDerivation`

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @peterhoeg

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/qgit/default.nix [hunk 1]
```diff
@@ -16,7 +17,7 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
```
Comment (peterhoeg) [inline pkgs/applications/version-management/git-and-tools/qgit/default.nix]: You can drop this as cmake enables it by default.

Comment (markuskowa) [reply pkgs/applications/version-management/git-and-tools/qgit/default.nix]: Nice feature! When was that introduced?

Comment (peterhoeg) [reply pkgs/applications/version-management/git-and-tools/qgit/default.nix]: Accoring to ```git blame```:

```
031138b7f02a (Orivej Desh            2017-12-02 06:20:07 +0000 110)     if ! [[ -v enableParallelBuilding ]]; then
031138b7f02a (Orivej Desh            2017-12-02 06:20:07 +0000 111)         enableParallelBuilding=1
031138b7f02a (Orivej Desh            2017-12-02 06:20:07 +0000 112)         echo "cmake: enabled parallel building"
031138b7f02a (Orivej Desh            2017-12-02 06:20:07 +0000 113)     fi
```

Comment (markuskowa) [reply pkgs/applications/version-management/git-and-tools/qgit/default.nix]: Thanks! That was quite a while ago. Good to know that this line is not necessary anymore for cmake derivations.

**Pull request merged**